File: Baekjoon/브루트 포스/1018 체스판 다시 칠하기.cpp
#include <iostream>
#include <string>
using namespace std;
int main() {
	char board[50][50];
	int n, m;
	int min = 64, wcnt = 0, bcnt = 0;
	cin >> n >> m;
	for (int i = 0; i < n; i++)
		cin >> chess[i];
	for (int i = 0; i < n - 7; i++) {
		for (int j = 0; j < m - 7; j++) {
		    int w = 0, b = 0
			for (int k = 0; k < 8; k++) {
				for (int l = 0; l < 8; l++) {
					if ((k + l) % 2) {
						if (chess[i + k][j + l] == 'W')
							w++;
						else
							b++;
					}
					else {
						if (chess[i + k][j + l] == 'B')
							w++;
						else
							b++;
					}
				}
			}
			if (min > w)min = w;
			if (min > b)min = b;
		}
	}
	cout << min;
}

File: Baekjoon/브루트 포스/1018 체스판 다시 칠하기.py
n, m = map(int, input().split())
board = []
result = 64
for i in range(n):
    board.append(input())

for i in range(n - 7):
    for j in range(m - 7):
        w, b = 0, 0
        for x in range(i, i + 8):
            for y in range(j, j + 8):
                if (x + y) % 2 == 0:
                    if board[x][y] == 'W':
                        b += 1
                    else:
                        w += 1
                else:
                    if board[x][y] == 'W':
                        w += 1
                    else:
                        b += 1
        result = min(w, b, result)

print(result)
